### DISCUSSION TITLE
v11.0.0-beta1

### DISCUSSION THREAD

**Discussion by BottlecapDave** [BottlecapDave/HomeAssistant-OctopusEnergy#857] [Announcements]
The next release has a few breaking changes, so wanted to get some more eyes on it before I release it to the masses. The main areas of focus are

* All attributes now display timestamps in local time. This will hopefully avoid any confusion during the BST months
* Refactored how off peak/peak sensors work to now better support 3 rate tariffs (e.g. Cosy) with an additional split between peak and standard rates and any future tariffs that may have 3 unique rates.
* Updates the default names to better support areas of HA with limited space.

The full release notes are below

### Bug Fixes

* fixed tariff override data having rate value in pounds and pence for consistency (48f832a (https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/commit/48f832afd9cb5ea846d655e7eb3ea7937520822b))
* Fixed updating cost tracker config (c934614 (https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/commit/c934614ccd0a9b5ee471280be2c0d565000a9c52))
* Updated attributes and events to have datetimes in local time (UTC in winter, BST in summer) (19ac60f (https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/commit/19ac60f10ae81f4a81bedbd448c5563c41e35805))

### Features

* Added ability to configure optional minimum/maximum rates for target rate sensors (95ac448 (https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/commit/95ac4489653a97cd2a3d3274908543d73ff5827f))
* Added support for tracking 3 rate tariffs (e.g. Cosy) via separate sensors (40d21d1 (https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/commit/40d21d1815ceb9f1e1eff1be7362b42b15950668))
* updated the default names of various sensors to be more useful in areas of HA when space is limited (870c917 (https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/commit/870c917fd4a166ea71a51971df9b98e7d75957bd))

### BREAKING CHANGES

* If you are relying on the previous default names of sensors then you will need to update your
  sensors manually. Only the visual default name of the sensors have changed; the entity ids will
  remain the same and therefore will not effect any automatmations. If you have manually changed the
  names of the sensors, this should not effect you.
* For 3 rate tariffs (e.g. Cosy), the peak based sensors will no longer track consumption/cost for both peak and standard rates. There is now a third sensor (standard) which tracks the 
middle rate. In addition, if you are on a tariff that doesn't have off peak, standard or peak rates, the current off peak/peak based sensors will no longer be provided by the integration.  
* If you were relying on datetime based sensor attributes or event properties being in UTC, you will
  need to adapt accordingly
* If you were using tariff override rate information, you may need to adjust your use due to changes
  in how value is reported

<hr /><em>This discussion was created from the release <a href='https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/releases/tag/v11.0.0-beta1'>v11.0.0-beta1</a>.</em>

**Comment by ssbarnea** [comment]
Some info about how to test this beta could be quite useful. I tried
 looking for the options described on https://hacs.xyz/docs/faq/beta/ but I was not able to find them.

I am quite keen to test the new beta due to the sensor renames. I just enabled it today so there is no risk of losing real tracking data if things do not go well.

**Reply by BottlecapDave** [reply]
Instructions can be adapted from a previous beta release - outlined at https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/issues/223#issuecomment-1902083426. This will be in the FAQ for future beta releases.
